### DISCUSSION TITLE
How WinUI protocol activation is handled on auth?

### DISCUSSION THREAD

**Discussion by xperiandri** [unoplatform/uno.extensions#2357] [Q&A] [answered]
I try to use OpenIddict Client instead of the current authentication package, so I try to understand how it works.
@nickrandolph can you point me to the right place in code. And maybe some generic WinUI sample.

**Comment by xperiandri** [comment, marked as answer]
OK, I've found that https://github.com/unoplatform/uno.extensions/blob/44e1018ec74cfebf8e0b81ff45799b3508fe50f4/src/Uno.Extensions.Authentication.UI/WinUIEx_WebAuthenticator.cs#L59

**Comment by DevTKSS** [comment]
@xperiandri hey there, found your discussion here, seems like you got that somehow working, could you help me implement this or do you have any sample to look into? my only reference is https://github.com/IlGalvo/WinUI3-OAuth2Manager-Sample which uses that protocol, but the api I am trying to reach is requiring https:// 🤔 is there something to know which I might miss, that would enable me to get such prefix for the callback of an oAuth2 uno app?

**Reply by xperiandri** [reply]
I use OpenIddict.Client instead

**Reply by DevTKSS** [reply]
@xperiandri do you have a uno using sample I could look into? Even tube player sample does just use api key auth so no pkce challenge 😅 but the api I need has no discovery Endpoint

**Reply by xperiandri** [reply]
https://github.com/Ecierge/openiddict-core/tree/uno

**Reply by xperiandri** [reply]
https://github.com/Ecierge/openiddict-core/tree/apple-cryptokit
